cmd/zrepl/zrepl.c

```diff
@@ -546,114 +602,120 @@ get_controller_ip_address(char *buf, int len)
 static void
 uzfs_zvol_mgmt_thread(void *arg)
 {
-	const char *target_addr = arg;
-	char ip_buf[256];
-	int rc;
-	struct sockaddr_in istgt_addr;
-	zvol_io_hdr_t hdr = {0, };
-	char *buf;
-	int sfd = -1;
+	int			rc;
+	char			*buf;
+	int			sfd = -1;
+	int			efd = -1;
+	uint32_t		flags;
+	zvol_io_hdr_t		hdr = {0, };
+	struct epoll_event	event;
+	struct epoll_event	*events = NULL;
+
+	efd = epoll_create1(0);
+	if (efd == -1) {
+		ZREPL_ERRLOG("epoll_create() failed with errno:%d\n", errno);
+		goto exit;
+	}
 
+	flags = EPOLLIN | EPOLLET | EPOLLERR | EPOLLHUP | EPOLLRDHUP;
 
-	sfd = create_and_bind(mgmt_port, false);
+	sfd = uzfs_zvol_connect_to_tgt_controller(arg);
```

Comment:
not sure whether I missed something, zrepl should be able to connect to multiple targets. How are we going to connect to multiple targets?

Reply:
To my surprise, this point has never came to discussion in past even on previous replication review request. Lets discuss more on it ....what we need and what should be the approach going farword.